mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -227,4 +230,19 @@ private JPAMailbox loadJpaMailbox(MailboxId id) throws MailboxNotFoundException
     public Flux<Mailbox> findNonPersonalMailboxes(Username userName, Right right) {
         return Flux.empty();
     }
+    
+    public List<Mailbox> findMailboxesForUser(MailboxPath mailboxPath) throws MailboxException {
+        try {
+            return getEntityManager().createNamedQuery("findMailboxByUserName", JPAMailbox.class)
+                    .setParameter("namespaceParam", mailboxPath.getNamespace())
+                    .setParameter("userParam", mailboxPath.getUser().asString())
```

Comment:
No it is different findMailboxByNameWithUser is considering nameParam but the new added method won't. for example if a user has 3 mailbox including Trash, Inbox and SENT the new added method loads all 3 and no need to pass mailbox name but findMailboxByPath only loads the one you provided the name

Reply:
Re read my comment please.

I speak of `findMailboxWithPathLike`.

Reply:
But the Name can be a SQL wildcard. Achieving what you want.

To be honnest, don't you think listing all the mailboxes of a user is somehow a pretty common task for a mail server and that somehow there should be a way to do it using the existing code? The contrary would be surprising....

Reply:
the requirement here is to load all user's mailbox.  let's assume a user has SENT, INBOX, TRASH mailboxes please enlighten me how do you do that with wildcard ? needless to say this query won't work 

```
select * from james_mailbox where user_name = 'auser@mydomain.tld' and mailbox_name like '%*%'
```

since this the query written in JPA.

Reply:
Can you provide an example that actually do not work?

Because I wrote the following test in `MailboxMapperTest` and ran it in JPA:

```

    @Test
    void proveFindMailboxWithPathLikeWorks() {
        Mailbox mailbox1 = createMailbox(MailboxPath.forUser(BENWA, "SENT"));
        Mailbox mailbox2 = createMailbox(MailboxPath.forUser(BENWA, "TRASH"));
        Mailbox mailbox3 = createMailbox(MailboxPath.forUser(BENWA, "INBOX"));

        Flux<Mailbox> mailboxes = mailboxMapper.findMailboxWithPathLike(MailboxQuery.builder()
            .userAndNamespaceFrom(benwaInboxPath)
            .expression(Wildcard.INSTANCE)
            .build()
            .asUserBound());
        List<String> names = mailboxes.map(Mailbox::getName).collectList().block();
        assertThat(names).containsOnly("SENT", "TRASH", "INBOX");
    }
```

Works...

So functionally speaking, it does the job...

Reply:
You are right. This code works. Wildcard.INSTANCE is generating this query 

```
select * from james_mailbox where user_name = 'user@mydomain.tld' and mailbox_name like '%'
```

and it loads all mailbox for the given user. 

Thanks. feel free to close this PR. to be honest that was complicated :D

Reply:
To be honnest working with the MailboxManager should feel easier and MailboxQuery have a builder that helps things up.

Happy to have had helped you.